### DISCUSSION TITLE
[Feature Request]: Support for binarized file formats like AVRO, ORC, Parquet etc.

### DISCUSSION THREAD

**Discussion by chitralverma** [schemacrawler/SchemaCrawler#580] [Q&A]
### Feature Request

Hello,
I would like raise a feature request to support binarized file formats like AVRO, ORC, Parquet etc.

With advent of big data, such optimized formats are extremely popular now and the make this library all inclusive, inclusion of these formats seems important.

### Possible Alternatives

Since these schema crawling for these formats is not done via JDBC, either these formats can be supported as their own sub modules or an API can be extended to supported these externally.

**Comment by sualeh** [comment]
@chitralverma - this is a great question, and the short answer is that this functionality may already be supported!

SchemaCrawler is designed to work with any compliant JDBC driver, and has an API designed to work with a relational structure.

[cdata](https://www.cdata.com/jdbc/) has developed a number of JDBC drivers that can make the internal schema of many systems (big data, noSQL, marketing and collaboration, CRM, ERP, e-commerce systems) available in the form of a relational schema via JDBC drivers. You can obtain the cdata JDBC drivers for Avro, Parquet, Hive and so on, and use them with SchemaCrawler.

I have not tried all of the cdata drivers - just the Salesforce one, and it works well.
